### PR TITLE
CHI-2614: Fix issue with the save and end button

### PR DESCRIPTION
**Pull request by acedeywin**
## Description

- Bug fix: Error when clicking the `Save and end` button multiple times.

- Root cause: This bug can be reproduced only when trying to save online contact. This is because the `completeTask` `async` function takes more time to resolve while saving online contact. By the time `submitContactFormAsyncAction` gets resolved, the `loadingStatus` is updated to loaded and as such enables the "Save and End" button, which can now be clicked, thus causing the error.

- Fix: To fix this, we need to always set the `loadingStatus` to loading and keep the "Save and end" button in a loading state until `completeTask` gets resolved.

### Checklist
- [ ] Corresponding issue has been opened
- [ ] New tests added
- [ ] Feature flags added
- [ ] Strings are localized
- [ ] Tested for chat contacts
- [ ] Tested for call contacts

### Other Related Issues

None

### Verification steps

- Start a web chat
- Accept the task as a counsellor
- Fill in the necessary details and click on the `Save and end` button
- Verify that the `Save and end` button cannot be clicked a second time while contact is been saved.

### CODE REVIEW

**Review by stephenhand — COMMENTED** [review]
How does the status get set back to 'LOADED' after the task work is completed? It seems like this would leave the contact in a LOADING state forever?

**Comment by acedeywin** [comment]
@stephenhand - Update has been implemented.

File: plugin-hrm-form/src/states/contacts/saveContact.ts [hunk 1]
```diff
@@ -243,6 +243,10 @@ export const loadContactIntoRedux = (
       ...existingContacts,
       [contact.id]: {
         ...existingContacts[contact.id],
+        /*
```
Comment (stephenhand) [inline plugin-hrm-form/src/states/contacts/saveContact.ts:246]: Need to remove this comment now?

**Review by stephenhand — APPROVED** [review]
1 clean up comment, otherwise LGTM

**Comment by nickhurlburt** [comment]
CHI-2591

**Pull request merged**